src/blockchain_db/sqlite/db_sqlite.cpp

```diff
@@ -172,6 +172,47 @@ namespace cryptonote {
 
       transaction.commit();
     }
+
+    bool have_archive = false;
+    SQLite::Statement archive_cols{db, "PRAGMA main.table_info(batched_payments_accrued_archive)"};
+    while (archive_cols.executeStep()) {
+      have_archive = true;
+      break;
+    }
+    if(!have_archive)
+    {
+      MINFO("Adding archiving to batching db");
+      auto& netconf = get_config(m_nettype);
+      SQLite::Transaction transaction{
+        db,
+        SQLite::TransactionBehavior::IMMEDIATE
+      };
+      db.exec(fmt::format(R"(
+        CREATE TABLE batched_payments_accrued_archive(
+          address VARCHAR NOT NULL,
+          amount BIGINT NOT NULL,
+          payout_offset INTEGER NOT NULL,
+          archive_height BIGINT NOT NULL,
+          PRIMARY KEY(address, archive_height),
+          CHECK(amount >= 0)
+        );
+
+        CREATE INDEX batched_payments_accrued_archive_height_idx ON batched_payments_accrued_archive(archive_height);
```

Comment:
You can drop this index if you reverse the order of the PRIMARY KEY, i.e. `PRIMARY KEY(archive_height, address)` (the leading columns of a multi-column index are usable as an index).

Reply:
Actually, I don't think we need the primary key at all (and it's going to be quite big because of the 95-char varchar in it), so we can keep this index and drop the PK.